### PR TITLE
Metrics: Distinguish internal and downstream erros in Grafana

### PR DESCRIPTION
**Pull request by bergquist**
This is POC of how we can change the errorsource label for `http_metrics` based on where the error came from.

### CODE REVIEW

File: pkg/middleware/request_metrics.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ func init() {
 			Help:      "Histogram of latencies for HTTP requests.",
 			Buckets:   defBuckets,
 		},
-		[]string{"handler", "status_code", "method"},
+		[]string{"handler", "status_code", "method", "errorSource"},
```
Comment (hairyhenderson) [inline pkg/middleware/request_metrics.go:44]:
```suggestion
		[]string{"handler", "status_code", "method", "error_source"},
```

File: pkg/api/http_server.go [hunk 2]
```diff
@@ -581,9 +581,9 @@ func (hs *HTTPServer) addMiddlewaresAndStaticRoutes() {
 	m := hs.web
 
 	m.Use(middleware.RequestTracing(hs.tracer))
-	m.Use(middleware.RequestMetrics(hs.Features))
 
 	m.UseMiddleware(middleware.Logger(hs.Cfg))
+	m.UseMiddleware(middleware.RequestMetrics(hs.Features))
```
Comment (hairyhenderson) [inline pkg/api/http_server.go:586]: I'm curious why this is getting moved to after the logger middleware?

Comment (bergquist) [reply pkg/api/http_server.go:586]: Indeed! I had to make this change in order to make it work locally so I suspect I hit a bug that I haven't looked deeper into.

File: pkg/middleware/request_metrics.go (hunk 1 above)
Comment (bergquist) [reply pkg/middleware/request_metrics.go:44]: 👍 

I also don't like the name in general but it's the best I've come up with so far. Perhaps `status_source` is better?

Comment (hairyhenderson) [reply pkg/middleware/request_metrics.go:44]: if it's populated for all statuses, maybe `status_source` makes sense...

Comment (hairyhenderson) [reply pkg/middleware/request_metrics.go:44]: That is to say, it's _only_ populated for errors. Should we populate this all the time? Or does that even make sense?

Comment (bergquist) [reply pkg/middleware/request_metrics.go:44]: `response_source` might be another alternative....

`status_source` feels like a good name when there is not an error. 

`error_source` feels better when the error comes from downtream services. 

🤔

**Comment by bergquist** [comment]
Closing this since someone else is picking it up

**Pull request closed without merging**